btree/btree.c

```diff
@@ -3346,6 +3346,58 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
  *	Pros: The performance can be slightly better as it will remove CRC
  *	storage and calculation.
  *	Cons: Any metadata corruption will result in undefined behavior.
+ *
+ *
+ * @brief Proposed design to use benefits of indirect addressing and embedding
+ *        records in nodes of FKVV type.
+ *
+ * In the case of FKVV with the embedded records, we store the key and value on
+ * the left and right locations in the node. But, while performing a write
+ * operation, we need to capture all the records which we moved in the node. To
+ * reduce the movement of records and transaction capturing, we design FKVV node
+ * format which provided embedded keys and values with indirect addressing.
+ *
+ * We will use following directory structure for key :
+ * struct dir_key {
+ *     uint32_t  validate;
+ *     uint32_t  val_size;
+ *     ---------- constant space for key ------------
+ *     void     *p_val;
+ *     };
+ *
+ *  +--+------+------+--+------+------+--+-----+-----+-------------------+--------+--------+--------+
+ *  |  |      |      |  |      |      |  |     |     |                   |        |        |        |
+ *  |  |      |      |  |      |      |  |     |     |                   |        |        |        |
+ *  |  |  k0  |P_V0  |  | K1   | P_V1 |  |  K2 | P_V2| ---->             |  V2    |  V1    |   V0   |
+ *  | ^|      |      |  |      |      |  |     |     |             <---- |        |        |        |
+ *  | ||      |      |  |      |      |  |     |     |                   |        |        |        |
+ *  | ||      |      |  |      |      |  |     |     |                   |        |        |        |
+ *  +-++------+--+---+--+------+--+---+--+-----+--+--+-------------------+----^---+-----^--+----^---+
+ *    |          |                |               |                           |         |       |
+ *    |          |                |               +---------------------------+         |       |
+ *    |          |                +---------------+-------------------------------------+       |
+ *    |          +------------------------------------------------------------------------------+
```

Comment:
80 char limit, can you remove extra spaces in between columns?

Reply:
Also what does first column in diagram indicate?? Node Header??? Please correct the straight line in first column.

Reply:
Updated diagram